crates/api/src/post/get_link_metadata.rs

```diff
@@ -14,6 +15,7 @@ use url::Url;
 pub async fn get_link_metadata(
   data: Query<GetSiteMetadata>,
   context: Data<LemmyContext>,
+  _local_user_view: LocalUserView,
```

Comment:
Is the unused param what handles adding the auth requirement?

Reply:
Yep. The auth middleware will throw an `invalid_auth` error if its not an `Option<LocalUserView>`

Reply:
here's the logic for that: https://github.com/LemmyNet/lemmy/blob/128e78f7c22279fec27d3213a05b440cb1f8db59/crates/db_views/src/local_user_view.rs#L128-L138